File: app/src/main/java/com/wa2c/android/medoly/plugin/action/lrclyrics/receiver/ToastReceiver.java
package com.wa2c.android.medoly.plugin.action.lrclyrics.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.widget.Toast;

/**
 * Toast receiver.
 */
public class ToastReceiver extends BroadcastReceiver {
    public static final String MESSAGE_TOAST = "message";

    @Override
    public void onReceive(Context context, Intent intent) {
        Toast.makeText(context, intent.getStringExtra(MESSAGE_TOAST), Toast.LENGTH_SHORT).show();
    }

    public static void showToast(Context context, int stringId) {
        Intent intent = new Intent(context, ToastReceiver.class);
        intent.putExtra(MESSAGE_TOAST, context.getString(stringId));
        context.sendBroadcast(intent);
    }

    public static void showToast(Context context, String text) {
        Intent intent = new Intent(context, ToastReceiver.class);
        intent.putExtra(MESSAGE_TOAST, text);
        context.sendBroadcast(intent);
    }

}

File: app/src/main/java/com/wa2c/android/medoly/plugin/action/lrclyrics/util/ToastReceiver.kt
package com.wa2c.android.medoly.plugin.action.lrclyrics.util

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.support.v4.content.ContextCompat
import com.pranavpandey.android.dynamic.toasts.DynamicToast
import com.wa2c.android.medoly.plugin.action.lrclyrics.R
import timber.log.Timber

/**
 * Toast receiver.
 */
class ToastReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        //Toast.makeText(context, intent.getStringExtra(MESSAGE_TOAST), Toast.LENGTH_SHORT).show()
        DynamicToast.make(context, intent.getStringExtra(MESSAGE_TOAST), ContextCompat.getDrawable(context, R.drawable.ic_launcher)).show()
    }

    companion object {
        private const val MESSAGE_TOAST = "message"

        fun showToast(context: Context, stringId: Int) {
            showToast(context, context.getString(stringId))
        }

        fun showToast(context: Context, text: String) {
            val intent = Intent(context, ToastReceiver::class.java)
            intent.putExtra(MESSAGE_TOAST, text)
            context.sendBroadcast(intent)
            Timber.d(text)
        }
    }

}
